CentralUnit/RaceManagement/RaceTracker.cs

```diff
@@ -228,15 +262,20 @@ private void OnTimer(TimingTriggeredEventArgs args)
 
                 if (hasWaitingRider)
                 {
-                    EnteredEvent rider = waitingRiders.Dequeue();
+                    IdEvent rider = waitingRiders.First();
+                    waitingRiders.RemoveFirst();
+
                     TimingEvent newEvent = new TimingEvent(args.Received, rider.Rider, args.Microseconds, args.GateId);
 
                     onTrackRiders.Enqueue((rider, newEvent));
                     raceState.Enqueue(newEvent);
 
+                    startGate.RemoveKnownRider(rider.Rider.Name);
```

Comment:
Why remove a rider from the start gate and add the rider back to the start gate at the end?
Can't we leave the rider known to the start gate or let a 'filter class' decide if a rider is added back to the start gate to allow for more flexibility?

Reply:
We remove it so we don't get bothered by rider events if the track gets close to the start box again.

In the future whether or the rider is added back should be decided by some kind of behviour (free riding -> add back, heat -> do not add back). But right now i think its best to assume people on the outide know they shouldn't drive back to the start box during a race

Reply:
Ok, seems like a reasonable assumption for now. Create an issue to add this behavior later?